decoder/src/stream/rzcobs.rs

```diff
@@ -0,0 +1,95 @@
+use super::StreamDecoder;
+use crate::{DecodeError, Frame, Table};
+
+/// Decode a full message.
+///
+/// `data` must be a full rzCOBS encoded message. Decoding partial
+/// messages is not possible. `data` must NOT include any `0x00` separator byte.
+fn rzcobs_decode(data: &[u8]) -> Result<Vec<u8>, DecodeError> {
+    let mut res = vec![];
+    let mut data = data.iter().rev().cloned();
+    while let Some(x) = data.next() {
```

Comment:
isn't there a crate we could depend on for this decoding part?

Reply:
Yes, this is from the `rzcobs` crate. For encoding it's not quite right, for decoding we could use it as-is. I thought it was strange to have an in-tree impl for encoding and a crate for decoding, but can change it if you think it's better to use external crates.